app/src/main/java/de/tum/in/tumcampusapp/managers/CardManager.java

```diff
@@ -56,10 +56,10 @@ public static void addCard(Card card) {
         if (card.getPosition() == -1) {
             card.setPosition(newCards.size());
         }
-
-        newCards.add(card);
-
-        Collections.sort(newCards);
+        synchronized (newCards) {
+            newCards.add(card);
+            Collections.sort(newCards);
```

Comment:
Ok, forget what I wrote.

Please don't sort the list every time a new card is added, but only once in the end.

Reply:
Well that's the thing, I can't, because the cafeteria card is added asynchronously now and we never  know when a card's going to be added or how many there'll be in the end.

Reply:
Then we probably need a real concurrent data structure. A `ConcurrentSkipListSet` seems to keep the sorting when adding...